### PR TITLE
DO_NOT_MERGE template service broker: use generated listers

### PR DESCRIPTION
**Pull request by jim-minter**
@deads2k @ncdc please could you take a look at this PR which builds on my last trello card commit (currently in merge queue), moving to use the generated lister and removes the old shared informer for templates.

It "works for me", but I think the template broker API server should somehow hold incoming API calls until cache sync has happened.  I don't know how to make this happen and need specific advice.  I don't know if this was a bug in the previous version of the code as well and I didn't realise it.

cc @bparees

### CODE REVIEW

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -454,12 +454,14 @@ func (m *Master) Start() error {
 			openshiftConfig.Informers.Start(utilwait.NeverStop)
 			openshiftConfig.Informers.StartCore(utilwait.NeverStop)
 			openshiftConfig.AuthorizationInformers.Start(utilwait.NeverStop)
+			openshiftConfig.TemplateInformers.Start(utilwait.NeverStop)
```
Comment (ncdc) [inline pkg/cmd/server/start/start_master.go]: You could run this via a goroutine, and add a line immediately after this one that waits for the caches to sync (blocking). WDYT @deads2k @sttts?

Comment (deads2k) [inline pkg/cmd/server/start/start_master.go]: Do we care about waiting?  As I recall, start is pretty quick anyway.

Comment (ncdc) [inline pkg/cmd/server/start/start_master.go]: I think @jim-minter wanted to make sure the caches were primed before any api requests were allowed

Comment (deads2k) [inline pkg/cmd/server/start/start_master.go]: > I think @jim-minter wanted to make sure the caches were primed before any api requests were allowed

We've done that at point of use before.  Not everything needs to wait for a full sync and certainly not our apiserver as a whole.

Comment (ncdc) [inline pkg/cmd/server/start/start_master.go]: ack, reasonable

**Pull request closed without merging**